Makefile

```diff
@@ -1,37 +1,13 @@
 all:
 
-PROG = bill
+compiler := src/compiler
 
-CXXFLAGS+= -std=c++17
-CXXFLAGS+= -Wall
-CXXFLAGS+= -Werror
-CXXFLAGS+= -O3
-#CXXFLAGS+= -lboost_unit_test_framework		# these lines required for test
-#LIBS+= boost/test/included/unit_test.hpp
-#LIBS+= boost_unit_test
+DIRS+=$(compiler)
 
-#unit_test.o: unit_test.hpp
-#	g++ -o unit_test.o -c boost/test/unit_test.hpp ${CXXFLAGS}
+TARGETS = all clean cleaner cleanest
+.PHONY: $(TARGETS) $(DIRS)
+$(TARGETS): $(DIRS)
```

Comment:
Make magic.  Every valid target depends on all the directories to be (sub)make-d.  We then give the actual command goals in the target to build each directory.